sigopt/objects.py

```diff
@@ -246,16 +246,27 @@ def iterate_pages(self):
     # pylint: disable=no-member
     data = self.data
     paging = self.paging or Paging({})
-    use_before = bool(paging.before)
+
+    # TODO(patrick): Probably should always rely on _retrieve_params vs. using paging.before,
+    # but that is divergent behaviour. Consider for next major version bump
+    specified_direction = ('before' in self._retrieve_params) ^ ('after' in self._retrieve_params)
+    if specified_direction:
+      use_before = 'before' in self._retrieve_params or 'after' not in self._retrieve_params
+    else:
+      use_before = bool(paging.before)
+
     while data:
       for d in data:
         yield d
       next_paging = dict(before=paging.before) if use_before else dict(after=paging.after)
       if next_paging.get('before') is not None or next_paging.get('after') is not None:
```

Comment:
Since we're not using `next_paging`, we could probably make this into a flag somehow (or go back to using it below, I think it should be OK now)